tests/samplers_tests/test_nsga2.py

```diff
@@ -190,8 +405,15 @@ def test_reseed_rng() -> None:
     assert original_random_sampler_seed != sampler._random_sampler._rng.seed
 
 
+def test_constraints_func_experimental_warning() -> None:
+    with pytest.warns(optuna.exceptions.ExperimentalWarning):
+        NSGAIISampler(constraints_func=lambda _: [0])
+
+
 # TODO(ohta): Consider to move this utility function to `optuna.testing` module.
-def _create_frozen_trial(number: int, values: List[float]) -> optuna.trial.FrozenTrial:
+def _create_frozen_trial(
+    number: int, values: List[float], constraints: Optional[List[float]] = None
+) -> optuna.trial.FrozenTrial:
```

Comment:
This can be done in the follow-up, but how about re-defining `_create_frozen_trial` by using `optuna.trial.create_trial` since the direct instantiation of `FrozenTrial` is not a recommended usage?

Reply:
That makes sense. I update the `_create_frozen_trial` according to your suggestion in https://github.com/optuna/optuna/pull/2175/commits/1040b8444c9df998d74a1373723c1082e86271e0. PTAL.